Training/configs/training_v1.yaml

```diff
@@ -33,248 +30,371 @@ CellObjectType : [
+#####       The corresponding `lim_min` and `lim_max` to meet the DataLoader notation are derived automatically in the function `init_dictionaries()`` of scaling_utils.py
+#####   - Furthermore, for the "normal" case, one should pass `lim_params` as the number of sigmas to which they want to clamp the data after being scaled by mean and std (e.g. [-5, 5])
+#####       It is also possible to skip `lim_params` argument. In that case, it is automatically filled `lim_min=-inf`, `lim_max=inf`
+#####   - Lastly, `lim_params` (for both linear and normal cases) can be a dictionary with keys "inner" and/or "outer" and values as lists of two elements as before.
+#####       In that case `lim_min`/`lim_max` will be derived separately for each specified cone.
+
 Features_all :
-    TauFlat : [ "tau_pt",
-                "tau_eta",
-                "tau_phi",
-                "tau_mass",
-                "tau_E_over_pt",
-                "tau_charge",
-                "tau_n_charged_prongs",
-                "tau_n_neutral_prongs",
-                "chargedIsoPtSum",
-                "chargedIsoPtSumdR03_over_dR05",
-                "footprintCorrection",
-                "neutralIsoPtSum",
-                "neutralIsoPtSumWeight_over_neutralIsoPtSum",
-                "neutralIsoPtSumWeightdR03_over_neutralIsoPtSum",
-                "neutralIsoPtSumdR03_over_dR05",
-                "photonPtSumOutsideSignalCone",
-                "puCorrPtSum",
-                "tau_dxy_valid",
-                "tau_dxy",
-                "tau_dxy_sig",
-                "tau_ip3d_valid",
-                "tau_ip3d",
-                "tau_ip3d_sig",
-                "tau_dz",
-                "tau_dz_sig_valid",
-                "tau_dz_sig",
-                "tau_flightLength_x",
-                "tau_flightLength_y",
-                "tau_flightLength_z",
-                "tau_flightLength_sig",
-                "tau_pt_weighted_deta_strip",
-                "tau_pt_weighted_dphi_strip",
-                "tau_pt_weighted_dr_signal",
-                "tau_pt_weighted_dr_iso",
-                "tau_leadingTrackNormChi2",
-                "tau_e_ratio_valid",
-                "tau_e_ratio",
-                "tau_gj_angle_diff_valid",
-                "tau_gj_angle_diff",
-                "tau_n_photons",
-                "tau_emFraction",
-                "tau_inside_ecal_crack",
-                "leadChargedCand_etaAtEcalEntrance_minus_tau_eta"
-                ]
+    TauFlat :
+                - tau_pt: [null, null, "linear", 20, 1000]
+                - tau_eta: [null, null, "linear", -2.3, 2.3]
+                - tau_phi: [null, null, "linear", *mpi, *pi]
+                - tau_mass: [null, null, "normal", -5, 5]
+                - tau_E_over_pt: [null, {"tau_E_over_pt": "sqrt((tau_pt*cosh(tau_eta))*(tau_pt*cosh(tau_eta)) + tau_mass*tau_mass)/tau_pt"}, "linear", 1., 5.2]
```

Comment:
Hm, I am thinking about this and from what I understand there might be essentially no difference between normal and linear scaling in those exceptions. Basically, they are both implemented as a linear transformation followed by clamping, and the only difference is how the range of clamping is specified in the training cfg. In linear case it is assumed to be "a priori known" (and hence explicitly written, like [1., 5.2]), while for normal scaling it is sort of not so we leave it to a suppossedly-large-enough-to-cover "+-5 std range". But then both of them might turn out to not entirely cover the whole distribution. For example, with linear case as it is in `tau_E_over_pt`, and for the normal case still 5 std can be not enough, because distributions are mostly not gaussian-like. So in the end there are still checks to be made with actual quantiles if we want to be sure. And if they reveal undercoverage for clamping, then for linear case it can be either adjusted by extending the range in cfg, or computing the mean/std and still adjusting the range but now in terms of std units, so it is basically the same thing :) That is why I am a bit confused, because we can simply respecify in the training cfg [1., 5.2] -> [1., 6.2] and that would do the job if excessive clamping is the concerning part.

Reply:
for the reference, this is how `tau_E_over_pt` distribution will look like after applying scaling with current parameters. Top: original distribution, bottom: transformed
![image](https://user-images.githubusercontent.com/20295221/121702388-1fc6d780-cad2-11eb-9362-5900d0e7b43a.png)

Reply:
and this is the corresponding quantile plot
![image](https://user-images.githubusercontent.com/20295221/121702915-9237b780-cad2-11eb-981d-9aa1012bd09d.png)

Reply:
Thank you for the details! I think, as a general practice, we should try to avoid having "custom" intervals in the config because it is less robust for possible changes. In `tau_E_over_p` case I think it is safe to switch to the normal transformation so that we don't need to manually control for each S&M production that the range is still relevant.

In the future, we can think about more automated range determination, because as you mentioned for some very non-gaussian distributions the (-5 * std, +5 * std) interval doesn't cover 95% of events, so clamping at 5 could be a bit suboptimal. But for the first round, I think we are fine with what we have in place + few minor tunings listed in the PR comments.

Reply:
updated in 2a1ec38